ui/src/shared/containers/SetOrg.tsx

```diff
@@ -0,0 +1,80 @@
+// Libraries
+import React, {useEffect, FunctionComponent} from 'react'
+import {connect} from 'react-redux'
+
+// Components
+import {SpinnerContainer, TechnoSpinner} from '@influxdata/clockface'
+
+// Types
+import {RemoteDataState, AppState} from 'src/types'
+import {Organization} from '@influxdata/influx'
+
+// Actions
+import {setOrg as setOrgAction} from 'src/organizations/actions/orgs'
+
+// Decorators
+import {InjectedRouter} from 'react-router'
+
+interface PassedInProps {
+  children: React.ReactElement<any>
+  router: InjectedRouter
+  params: {orgID: string}
+}
+
+interface DispatchProps {
+  setOrg: typeof setOrgAction
+}
+
+interface StateProps {
+  orgs: Organization[]
+  status: RemoteDataState
+}
+
+type Props = StateProps & DispatchProps & PassedInProps
+
+const GetOrganizations: FunctionComponent<Props> = ({
+  params: {orgID},
+  orgs,
+  router,
+  setOrg,
+  children,
+  status,
+}) => {
+  useEffect(() => {
+    // does orgID from url match any orgs that exist
+    const foundOrg = orgs.find(o => o.id === orgID)
+    const orgExists = !!orgID && !!foundOrg
+    if (orgExists) {
+      setOrg(foundOrg)
+      return
+    }
+
+    // else default to first org
+    router.push(`orgs/${orgs[0].id}`)
+  }, [orgID])
+
+  return (
+    <SpinnerContainer loading={status} spinnerComponent={<TechnoSpinner />}>
```

Comment:
No asynchronous work is happening here.  I don't think any of the loading or spinning states / logic / presentation is needed.

Reply:
actually, we don't want to render the children unless org has been set.